mailbox/event/json/src/main/scala/org/apache/james/event/json/DTOs.scala

```diff
@@ -74,75 +74,103 @@ object DTOs {
     }
 
-    def toJavaMailFlag(flag: String): JavaMailFlags = ALL_SYSTEM_FLAGS.contains(flag) match {
-      case true => new FlagsBuilder().add(stringToSystemFlag(flag)).build()
-      case false => new FlagsBuilder().add(flag).build()
+    def toScala(flag: JavaMailFlags.Flag): SystemFlag = flag match {
+      case JavaMailFlags.Flag.ANSWERED => Answered
+      case JavaMailFlags.Flag.DELETED => Deleted
+      case JavaMailFlags.Flag.DRAFT => Draft
+      case JavaMailFlags.Flag.FLAGGED => Flagged
+      case JavaMailFlags.Flag.RECENT => Recent
+      case JavaMailFlags.Flag.SEEN => Seen
     }
 
-    def fromJavaFlags(flags: JavaMailFlags): Array[String] = {
-      flags.getUserFlags ++ flags.getSystemFlags.map(flag => systemFlagToString(flag))
+    val ANSWERED = "Answered"
+    val DELETED = "Deleted"
+    val DRAFT = "Draft"
+    val FLAGGED = "Flagged"
+    val RECENT = "Recent"
+    val SEEN = "Seen"
+
+    def fromString(value: String): SystemFlag = value match {
+      case ANSWERED => Answered
+      case DELETED => Deleted
+      case DRAFT => Draft
+      case FLAGGED => Flagged
+      case RECENT => Recent
+      case SEEN => Seen
+      case unknown => throw new RuntimeException("Unknown system flag " + unknown)
     }
+  }
 
-    private def stringToSystemFlag(serializedFlag: String): JavaMailFlags.Flag = serializedFlag match {
-      case ANSWERED => JavaMailFlags.Flag.ANSWERED
-      case DELETED => JavaMailFlags.Flag.DELETED
-      case DRAFT => JavaMailFlags.Flag.DRAFT
-      case FLAGGED => JavaMailFlags.Flag.FLAGGED
-      case RECENT => JavaMailFlags.Flag.RECENT
-      case SEEN => JavaMailFlags.Flag.SEEN
-      case _ => throw new IllegalArgumentException(serializedFlag + " is not a system flag")
-    }
+  sealed trait SystemFlag {
+    def rawString: String
+  }
+  case object Answered extends SystemFlag {
+    override def rawString: String = SystemFlag.ANSWERED
+  }
+  case object Deleted extends SystemFlag {
+    override def rawString: String = SystemFlag.DELETED
+  }
+  case object Draft extends SystemFlag {
+    override def rawString: String = SystemFlag.DRAFT
+  }
+  case object Flagged extends SystemFlag {
+    override def rawString: String = SystemFlag.FLAGGED
+  }
+  case object Recent extends SystemFlag {
+    override def rawString: String = SystemFlag.RECENT
+  }
+  case object Seen extends SystemFlag {
+    override def rawString: String = SystemFlag.SEEN
+  }
 
-    private def systemFlagToString(flag: JavaMailFlags.Flag): String = flag match {
-      case JavaMailFlags.Flag.ANSWERED => ANSWERED
-      case JavaMailFlags.Flag.DELETED => DELETED
-      case JavaMailFlags.Flag.DRAFT => DRAFT
-      case JavaMailFlags.Flag.FLAGGED => FLAGGED
-      case JavaMailFlags.Flag.RECENT => RECENT
-      case JavaMailFlags.Flag.SEEN => SEEN
-    }
+  object Flags {
+    def toScala(flags: JavaMailFlags): Flags = Flags(
+        Option(flags.getSystemFlags).getOrElse(Array()).map(SystemFlag.toScala).toList,
```

Comment:
you don't need the `getOrElse` because Option can implicitly convert itself into a Seq, basically, None.toList is the empty list.